host-controller/src/main/java/org/jboss/as/domain/controller/transformers/DomainTransformers.java

```diff
@@ -66,11 +66,12 @@ public static void initializeDomainRegistry(final TransformerRegistry registry)
         //For JBoss EAP: 8.0.0 -> 5.0.0 -> 4.0.0 -> 1.8.0 -> 1.7.0 -> 1.6.0 -> 1.5.0
 
         registerRootTransformers(registry);
-        registerChainedManagementTransformers(registry);
-        registerChainedServerGroupTransformers(registry);
+        ChainedTransformationDescriptionBuilderFactory factory = KernelAPIVersion.createChainedTransformationDescriptionBuilderFactory(registry);
+        registerChainedManagementTransformers(registry, factory);
+        registerChainedServerGroupTransformers(registry, factory);
         registerProfileTransformers(registry);
-        registerSocketBindingGroupTransformers(registry);
-        registerDeploymentTransformers(registry);
+        registerSocketBindingGroupTransformers(registry, factory);
+        registerDeploymentTransformers(registry, factory);
```

Comment:
Why we are only changing the Domain Transformers? Shouldn't we also change and adapt in this PR how we are registering the other transformers?

For example, at ElytronSubsystemTransformers we have:

```
@Override
    public void registerTransformers(SubsystemTransformerRegistration registration) {
        ChainedTransformationDescriptionBuilder chainedBuilder = TransformationDescriptionBuilder.Factory.createChainedSubystemInstance(registration.getCurrentSubsystemVersion());
```

What would be the equivalent code to create `ChainedTransformationDescriptionBuilder` using the current Stability level?